src/Controls/src/Core/Compatibility/Handlers/Shell/iOS/ShellSectionRenderer.cs

```diff
@@ -466,11 +467,49 @@ protected virtual void UpdateTabBarItem()
 
 			ShellSection.Icon.LoadImage(ShellSection.FindMauiContext(), icon =>
 			{
-				TabBarItem = new UITabBarItem(ShellSection.Title, icon?.Value, null);
+				TabBarItem = new UITabBarItem(ShellSection.Title, GetTabBarItemIcon(icon?.Value), null);
 				TabBarItem.AccessibilityIdentifier = ShellSection.AutomationId ?? ShellSection.Title;
 			});
 		}
 
+		// https://developer.apple.com/design/human-interface-guidelines/components/navigation-and-search/tab-bars/
+		UIImage GetTabBarItemIcon(UIImage source)
+		{
+			CGSize imageSize = CGSize.Empty;
+
+			var scale = UIScreen.MainScreen.Scale;
+
+			if (scale == 3)
+				imageSize = new CGSize(51, 51);
+
+			if (scale == 2)
+				imageSize = new CGSize(34, 34);
+
+			return ResizeTabBarItemIcon(source, imageSize);
+		}
+
+		UIImage ResizeTabBarItemIcon(UIImage source, CGSize size)
```

Comment:
Is this the only way? Where should this live? Maybe this all goes away if people set a BaseSize in the `<MauiImage>` element?

Instead of generating images at runtime, should we use that feature - or maybe even add a feature to resizetizer to say what the use is for? We could for example have a `<MauiIMage Include="cat.svg" Intention="TabIcon" />` and then in each platform we calculate the defaults for properties - in the case of iOS, this "TabIcon" would translate to a multiple of 17 in iOS and a multiple of 23 in Mac Catalyst?